scripts/spec-alignment/differ.config.yaml

```diff
@@ -1,4 +1,4 @@
ea06bfd1477d2d","7e2550bab51f22b2","87ebd29bd49fc52f","ef39dd5223e0d080","fe14e7f0503ea188","6146023d4d5708a2","aebb8e4348b6d667","973a2e6704dbf3","fb4cac598a68c592","257c7eb81d6eb245","2a8e94fe037e13fd","5e7eb878de54eec6","960a9c64cd4ec93c","57dfd255520bd849","ec333ff8a708db69","1cc1ff39ad91ee69","5714652b88e2d44f","7a53b3b037c56325","8c02ef1964464c30","19a268910a20da3d","af6e01ed565029f3","318b5169ac4dabb6","372c6b332e8ba699","c0d8a364c0db855a","4287381be4fb1841","b1614afc1da7794f","c214975412f3fd7","8bbf7eba3fa0cf7e","8e4ec8debe331b36","7a671d8fcefc3793","e2b0e9c6454c1c08","6707ecfefa5fec21","d5a7389d730464f1","8dfae3b3223d2de0","a81c092c985de728","968df5082c727ed6","9e53c73ee60b1cc2","9d265e99dd31d4f5","a34619e078d2e42f","17e8cec4f0625d53","e913f373aa88f333","cfc1e05c372d88dc","e5de6901d78b8833","57c1885b43dd8d19","e8a49856a5edd893","22ea21d10a2f861c","954e4fce01631c4e","108b9575f7c1d4bc","1f8d076449068f64","5a7ad98296703f6","159536003eeddac8","8ca8f82e67ddd3dd","16ebe47404323cc1","48bfe5cf1e578b47","dd83182b693a7216","308d21d9830f7047","6dde03147e012b1a","730c3e5e59393b7d","5b44a4b425ecc397","df5debc50ec8babc","92a41554b2910bb8","c36c680554dde59f","447feaa5cdc1a010","fda90c61f44cb149","cdbb4930eced584c","274336ec1127e6c0","2a496f5b3ad542d2","6b395912dde33b0e","cac56ec14994216b","8850900b5d9bcc65","15e7706486c6359e","cc22f28953b787ea","3bad6ae11596a574","8f84422a240d889c","5b265432dfbbaac7","43794bf5953db193","7975821460ebe1e7","173c505e12aabb8f","47ee0d148148a56f","8cc38593ebe049b6","bda3aec7157b095a","248712911696a851","f4d9c910f1dbaef7","1a2146fcad37acb8","b0b146f9bdab64b6","edfd442b4d725fbb","122f053573538a32","d720d714a20833e1", "f9c984e71b685f9b","8c6b4fee5a4c13ce","c0a8d2019a2c30d5", "717bef26105c733f","2f70630c27062353","2f70337ba7566a69","dd607a44e1341e6b","5210501625ac3de5","f786bf475b5085aa","18a66ed6e613d9c1","e8943e7741f6843d","276a489bd5a00032","ba3bba59f10bf6b","3c50ce0c8089d871","89ee72f6c610ab84","c92b95a85da2cb11","927ea6aed3f98f20","9338904026a0ce37","9683cfa19dc544a3","4d3fa2b8dfcb5f5b", "f19e9a2b295bcfb3", "b10199b2de6f03b8", "1afc17e358f9ca79","4b58762c0b433442","d293ec1bc61bb707","3e88c3b49d093605","4890ff80c88cc41d","5227ff3a225dd20d","81a60407a3a0ba80","db2ad807eb66254a","d308bd7c553ccdcf","bdaf172971637cbe","6ade9202843071fe","2fe8e14083997744","19c9a5362d1e1d3a","5956f803d239f178","92c55a4548a8b760","9a95524213bccfff","2f51a7338b86c229","e96966a281d74505","3ee479b9cbbc3a1d","82b392ba39c6c594","b9d2404e5c570019","24f528d85fb021f2","fe9609a785305d81","b0934079dcd986cc","a9c520a19b26049","d19a9403fd732d94","74a928f5dcb2fdd9","cbbfdb5e68cdac80","10e39d2ceda91f34","f99a004cf6697875","8fa5e8ebf7d223ec","6c80c145ba705243","fbabbc90d0b4178a","ab0c7f24e551ca6","af38a11cb8682c75","b110cba51df9f8d2","c4ff2ed3d20dc419","9295a5bb10efcec7","ab56ea44a75f898a","ff51ef26ab53ba58","df3771e2589008f9","106e5689655bcfc6","f90e0fb6883bff93","667656095cec39ee","9a5597af260c748a", "b40d693202be4382", "9129f80a9f1b1331", "b3b73f9cdf1c39fb"]
```

Comment:
I assume this is what we do to satisfy `spec-alignment` pipeline, the diff generated for this PR looks like this:
```
diff --git a/protocol/v2/ssv/validator/committee.go@Committee b/ssv/committee.go@Committee
index b40d693202be4382..2222222
--- a/protocol/v2/ssv/validator/committee.go@Committee
+++ b/ssv/committee.go@Committee
@@ -1,14 +1,6 @@
 Committee struct {
-	logger	*zap.Logger
-	ctx	context.Context
-	cancel	context.CancelFunc
-	BeaconNetwork	BeaconNetwork
-	Storage		*storage.QBFTStores
-	mtx	sync.RWMutex
-	Queues	map[phase0.Slot]queueContainer
-	Runners	map[phase0.Slot]*runner.CommitteeRunner
-	Shares	map[phase0.ValidatorIndex]*Share
-	CommitteeMember	*CommitteeMember
-	dutyGuard	*CommitteeDutyGuard
-	CreateRunnerFn	CommitteeRunnerFunc
+	Runners		map[spec.Slot]*CommitteeRunner
+	CommitteeMember	CommitteeMember
+	CreateRunnerFn	CreateRunnerFn
+	Share		map[spec.ValidatorIndex]*Share
 }
\ No newline at end of file
diff --git a/protocol/v2/ssv/validator/committee.go@Committee.ProcessMessage b/ssv/committee.go@Committee.ProcessMessage
index 9129f80a9f1b1331..2222222
--- a/protocol/v2/ssv/validator/committee.go@Committee.ProcessMessage
+++ b/ssv/committee.go@Committee.ProcessMessage
@@ -1,55 +1,48 @@
-func (c *Committee) ProcessMessage(msg *queue.SSVMessage) error {
-	if msg.GetType() != message.SSVEventMsgType {
-		if err := msg.SignedSSVMessage.Validate(); err != nil {
-			return errors.Wrap(err, "invalid SignedSSVMessage")
-		}
-		if err := Verify(msg.SignedSSVMessage, c.CommitteeMember.Committee); err != nil {
-			return errors.Wrap(err, "SignedSSVMessage has an invalid signature")
-		}
-		if err := c.validateMessage(msg.SignedSSVMessage.SSVMessage); err != nil {
-			return errors.Wrap(err, "Message invalid")
-		}
+func (c *Committee) ProcessMessage(signedSSVMessage *SignedSSVMessage) error {
+	if err := signedSSVMessage.Validate(); err != nil {
+		return errors.Wrap(err, "invalid SignedSSVMessage")
+	}
+	if err := Verify(signedSSVMessage, c.CommitteeMember.Committee); err != nil {
+		return errors.Wrap(err, "SignedSSVMessage has an invalid signature")
+	}
+	msg := signedSSVMessage.SSVMessage
+	if err := c.validateMessage(msg); err != nil {
+		return errors.Wrap(err, "Message invalid")
 	}
 	switch msg.GetType() {
 	case SSVConsensusMsgType:
 		qbftMsg := &Message{}
 		if err := qbftMsg.Decode(msg.GetData()); err != nil {
 			return errors.Wrap(err, "could not get consensus Message from network Message")
 		}
 		if err := qbftMsg.Validate(); err != nil {
 			return errors.Wrap(err, "invalid qbft Message")
 		}
-		c.mtx.RLock()
-		r, exists := c.Runners[phase0.Slot(qbftMsg.Height)]
-		c.mtx.RUnlock()
+		runner, exists := c.Runners[spec.Slot(qbftMsg.Height)]
 		if !exists {
-			return fmt.Errorf("no runner found for consensus message's slot %d", qbftMsg.Height)
+			return errors.New("no runner found for message's slot")
 		}
-		return r.ProcessConsensus(msg.SignedSSVMessage)
+		return runner.ProcessConsensus(signedSSVMessage)
 	case SSVPartialSignatureMsgType:
 		pSigMessages := &PartialSignatureMessages{}
-		if err := pSigMessages.Decode(msg.SignedSSVMessage.SSVMessage.GetData()); err != nil {
+		if err := pSigMessages.Decode(msg.GetData()); err != nil {
 			return errors.Wrap(err, "could not get post consensus Message from network Message")
 		}
-		if len(msg.SignedSSVMessage.OperatorIDs) != 1 {
+		if len(signedSSVMessage.OperatorIDs) != 1 {
 			return errors.New("PartialSignatureMessage has more than 1 signer")
 		}
-		if err := pSigMessages.ValidateForSigner(msg.SignedSSVMessage.OperatorIDs[0]); err != nil {
+		if err := pSigMessages.ValidateForSigner(signedSSVMessage.OperatorIDs[0]); err != nil {
 			return errors.Wrap(err, "invalid PartialSignatureMessages")
 		}
 		if pSigMessages.Type == PostConsensusPartialSig {
-			c.mtx.RLock()
-			r, exists := c.Runners[pSigMessages.Slot]
-			c.mtx.RUnlock()
+			runner, exists := c.Runners[pSigMessages.Slot]
 			if !exists {
-				return fmt.Errorf("no runner found for post consensus sig message's slot %d", pSigMessages.Slot)
+				return errors.New("no runner found for message's slot")
 			}
-			return r.ProcessPostConsensus(pSigMessages)
+			return runner.ProcessPostConsensus(pSigMessages)
 		}
-	case message.SSVEventMsgType:
-		return c.handleEventMessage(msg)
 	default:
 		return errors.New("unknown msg")
 	}
 	return nil
 }
\ No newline at end of file
diff --git a/protocol/v2/ssv/validator/committee.go@Committee.StartDuty b/ssv/committee.go@Committee.StartDuty
index b3b73f9cdf1c39fb..2222222
--- a/protocol/v2/ssv/validator/committee.go@Committee.StartDuty
+++ b/ssv/committee.go@Committee.StartDuty
@@ -1,11 +1,24 @@
 func (c *Committee) StartDuty(duty *CommitteeDuty) error {
-	r, trimmedDuty, err := c.prepareDutyRunner(duty)
-	if err != nil {
-		return fmt.Errorf("could not prepare duty runner: %w", err)
+	if len(duty.ValidatorDuties) == 0 {
+		return errors.New("no beacon duties")
 	}
-	err = r.StartNewDuty(trimmedDuty, c.CommitteeMember.GetQuorum())
-	if err != nil {
-		return errors.Wrap(err, "runner failed to start duty")
+	if _, exists := c.Runners[duty.Slot]; exists {
+		return errors.New(fmt.Sprintf("CommitteeRunner for slot %d already exists", duty.Slot))
 	}
-	return nil
+	dutyShares := make(map[spec.ValidatorIndex]*Share)
+	filteredDuty := &CommitteeDuty{
+		Slot: duty.Slot,
+	}
+	for _, bduty := range duty.ValidatorDuties {
+		if _, exists := c.Share[bduty.ValidatorIndex]; !exists {
+			continue
+		}
+		dutyShares[bduty.ValidatorIndex] = c.Share[bduty.ValidatorIndex]
+		filteredDuty.ValidatorDuties = append(filteredDuty.ValidatorDuties, bduty)
+	}
+	if len(dutyShares) == 0 {
+		return errors.New("no shares for duty's validators")
+	}
+	c.Runners[filteredDuty.Slot] = c.CreateRunnerFn(dutyShares)
+	return c.Runners[filteredDuty.Slot].StartNewDuty(filteredDuty, c.CommitteeMember.GetQuorum())
 }
\ No newline at end of file
```

Not sure why it works like this - but doesn't feel very useful to have `spec-alignment` fail when implementation detail change. Perhaps we can change it somehow to only call out `func` / `interface` signature updates ?

Reply:
Updated spec alignment (after rebase) - https://github.com/ssvlabs/ssv/pull/1784/commits/7e529a404b07d724402c44b188dc44baa99329ac, new diff:
```
diff --git a/protocol/v2/ssv/validator/committee.go@Committee b/ssv/committee.go@Committee
index 4418aa6010632480..2222222
--- a/protocol/v2/ssv/validator/committee.go@Committee
+++ b/ssv/committee.go@Committee
@@ -1,13 +1,6 @@
 Committee struct {
-	logger	*zap.Logger
-	ctx	context.Context
-	cancel	context.CancelFunc
-	BeaconNetwork	BeaconNetwork
-	mtx	sync.RWMutex
-	Queues	map[phase0.Slot]queueContainer
-	Runners	map[phase0.Slot]*runner.CommitteeRunner
-	Shares	map[phase0.ValidatorIndex]*Share
-	CommitteeMember	*CommitteeMember
-	dutyGuard	*CommitteeDutyGuard
-	CreateRunnerFn	CommitteeRunnerFunc
+	Runners		map[spec.Slot]*CommitteeRunner
+	CommitteeMember	CommitteeMember
+	CreateRunnerFn	CreateRunnerFn
+	Share		map[spec.ValidatorIndex]*Share
 }
\ No newline at end of file
diff --git a/protocol/v2/ssv/validator/committee.go@Committee.ProcessMessage b/ssv/committee.go@Committee.ProcessMessage
index 4cf190dd314b064b..2222222
--- a/protocol/v2/ssv/validator/committee.go@Committee.ProcessMessage
+++ b/ssv/committee.go@Committee.ProcessMessage
@@ -1,55 +1,48 @@
-func (c *Committee) ProcessMessage(msg *queue.SSVMessage) error {
-	if msg.GetType() != message.SSVEventMsgType {
-		if err := msg.SignedSSVMessage.Validate(); err != nil {
-			return errors.Wrap(err, "invalid SignedSSVMessage")
-		}
-		if err := Verify(msg.SignedSSVMessage, c.CommitteeMember.Committee); err != nil {
-			return errors.Wrap(err, "SignedSSVMessage has an invalid signature")
-		}
-		if err := c.validateMessage(msg.SignedSSVMessage.SSVMessage); err != nil {
-			return errors.Wrap(err, "Message invalid")
-		}
+func (c *Committee) ProcessMessage(signedSSVMessage *SignedSSVMessage) error {
+	if err := signedSSVMessage.Validate(); err != nil {
+		return errors.Wrap(err, "invalid SignedSSVMessage")
+	}
+	if err := Verify(signedSSVMessage, c.CommitteeMember.Committee); err != nil {
+		return errors.Wrap(err, "SignedSSVMessage has an invalid signature")
+	}
+	msg := signedSSVMessage.SSVMessage
+	if err := c.validateMessage(msg); err != nil {
+		return errors.Wrap(err, "Message invalid")
 	}
 	switch msg.GetType() {
 	case SSVConsensusMsgType:
 		qbftMsg := &Message{}
 		if err := qbftMsg.Decode(msg.GetData()); err != nil {
 			return errors.Wrap(err, "could not get consensus Message from network Message")
 		}
 		if err := qbftMsg.Validate(); err != nil {
 			return errors.Wrap(err, "invalid qbft Message")
 		}
-		c.mtx.RLock()
-		r, exists := c.Runners[phase0.Slot(qbftMsg.Height)]
-		c.mtx.RUnlock()
+		runner, exists := c.Runners[spec.Slot(qbftMsg.Height)]
 		if !exists {
-			return fmt.Errorf("no runner found for consensus message's slot %d", qbftMsg.Height)
+			return errors.New("no runner found for message's slot")
 		}
-		return r.ProcessConsensus(msg.SignedSSVMessage)
+		return runner.ProcessConsensus(signedSSVMessage)
 	case SSVPartialSignatureMsgType:
 		pSigMessages := &PartialSignatureMessages{}
-		if err := pSigMessages.Decode(msg.SignedSSVMessage.SSVMessage.GetData()); err != nil {
+		if err := pSigMessages.Decode(msg.GetData()); err != nil {
 			return errors.Wrap(err, "could not get post consensus Message from network Message")
 		}
-		if len(msg.SignedSSVMessage.OperatorIDs) != 1 {
+		if len(signedSSVMessage.OperatorIDs) != 1 {
 			return errors.New("PartialSignatureMessage has more than 1 signer")
 		}
-		if err := pSigMessages.ValidateForSigner(msg.SignedSSVMessage.OperatorIDs[0]); err != nil {
+		if err := pSigMessages.ValidateForSigner(signedSSVMessage.OperatorIDs[0]); err != nil {
 			return errors.Wrap(err, "invalid PartialSignatureMessages")
 		}
 		if pSigMessages.Type == PostConsensusPartialSig {
-			c.mtx.RLock()
-			r, exists := c.Runners[pSigMessages.Slot]
-			c.mtx.RUnlock()
+			runner, exists := c.Runners[pSigMessages.Slot]
 			if !exists {
-				return fmt.Errorf("no runner found for post consensus partial-sig message's slot %d", pSigMessages.Slot)
+				return errors.New("no runner found for message's slot")
 			}
-			return r.ProcessPostConsensus(pSigMessages)
+			return runner.ProcessPostConsensus(pSigMessages)
 		}
-	case message.SSVEventMsgType:
-		return c.handleEventMessage(msg)
 	default:
 		return errors.New("unknown msg")
 	}
 	return nil
 }
\ No newline at end of file
diff --git a/protocol/v2/ssv/validator/committee.go@Committee.StartDuty b/ssv/committee.go@Committee.StartDuty
index b3b73f9cdf1c39fb..2222222
--- a/protocol/v2/ssv/validator/committee.go@Committee.StartDuty
+++ b/ssv/committee.go@Committee.StartDuty
@@ -1,11 +1,24 @@
 func (c *Committee) StartDuty(duty *CommitteeDuty) error {
-	r, trimmedDuty, err := c.prepareDutyRunner(duty)
-	if err != nil {
-		return fmt.Errorf("could not prepare duty runner: %w", err)
+	if len(duty.ValidatorDuties) == 0 {
+		return errors.New("no beacon duties")
 	}
-	err = r.StartNewDuty(trimmedDuty, c.CommitteeMember.GetQuorum())
-	if err != nil {
-		return errors.Wrap(err, "runner failed to start duty")
+	if _, exists := c.Runners[duty.Slot]; exists {
+		return errors.New(fmt.Sprintf("CommitteeRunner for slot %d already exists", duty.Slot))
 	}
-	return nil
+	dutyShares := make(map[spec.ValidatorIndex]*Share)
+	filteredDuty := &CommitteeDuty{
+		Slot: duty.Slot,
+	}
+	for _, bduty := range duty.ValidatorDuties {
+		if _, exists := c.Share[bduty.ValidatorIndex]; !exists {
+			continue
+		}
+		dutyShares[bduty.ValidatorIndex] = c.Share[bduty.ValidatorIndex]
+		filteredDuty.ValidatorDuties = append(filteredDuty.ValidatorDuties, bduty)
+	}
+	if len(dutyShares) == 0 {
+		return errors.New("no shares for duty's validators")
+	}
+	c.Runners[filteredDuty.Slot] = c.CreateRunnerFn(dutyShares)
+	return c.Runners[filteredDuty.Slot].StartNewDuty(filteredDuty, c.CommitteeMember.GetQuorum())
 }
\ No newline at end of file
```